packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -422,6 +449,7 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
             await Resource.save(this.resource, { changes, content, options: { encoding, overwriteEncoding, version } }, token);
             this.contentChanges.splice(0, changes.length);
             this.resourceVersion = this.resource.version;
+            this.updateContentEncoding();
```

Comment:
Set a preferred encoding is always called with overwrite true, so it could happen only if there is then a concurrent save which cancels overwrite save. It was also bothering me on master, but I decided don't change it since did not want to refactor much here. I will file a follow-up issue. I don't think it is caused by this PR.